File: frontend/src/api/industreamTsd.ts
/**
 * Industream TSD — API client.
 *
 * Clean START/END pairing with machine transactions (2.har verified).
 */

import { apiClient } from './client'

export interface TsdStartRequest {
  job: string
  suffix?: string
  oper_num: string
  wc?: string
  item?: string
  whse?: string
}

export interface TsdEndWorkRequest extends TsdStartRequest {
  qty_complete?: number
  qty_scrapped?: number
  oper_complete?: boolean
  job_complete?: boolean
}

export interface TsdResponse {
  status: string
  message: string
  actual_hours?: number
  stroj?: string
  wc?: string
}

const BASE = '/industream-tsd'

/** Start Setup — WrapperSp(TT=1) + local tx */
export async function startSetup(req: TsdStartRequest): Promise<TsdResponse> {
  const { data } = await apiClient.post<TsdResponse>(`${BASE}/start-setup`, req)
  return data
}

/** End Setup + auto-Start Work — WrapperSp(TT=2) → full start_work flow */
export async function endSetup(req: TsdStartRequest): Promise<TsdResponse> {
  const { data } = await apiClient.post<TsdResponse>(`${BASE}/end-setup`, req)
  return data
}

/** Start Work — ValidateMachine + Wrapper(TT=3) + Kapacity + Mchtrx */
export async function startWork(req: TsdStartRequest): Promise<TsdResponse> {
  const { data } = await apiClient.post<TsdResponse>(`${BASE}/start-work`, req)
  return data
}

/** End Work — InsWrapper(TT=4) + Kapacity + MachineVal + DcSfcMchtrx */
export async function endWork(req: TsdEndWorkRequest): Promise<TsdResponse> {
  const { data } = await apiClient.post<TsdResponse>(`${BASE}/end-work`, req)
  return data
}

File: app/schemas/industream_tsd.py
"""Industream TSD — Request/Response schemas.

Clean START/END pairing with machine transactions (2.har verified).
"""

from typing import Optional

from pydantic import BaseModel, Field


class TsdStartRequest(BaseModel):
    """Request for start-setup, end-setup, start-work."""

    job: str = Field(..., min_length=1, max_length=30)
    suffix: str = Field(default="0", max_length=5)
    oper_num: str = Field(..., min_length=1, max_length=10)
    wc: str = Field(default="", max_length=20, description="Work center")
    item: str = Field(default="", max_length=30, description="DerJobItem")
    whse: str = Field(default="MAIN", max_length=10, description="Warehouse")


class TsdEndWorkRequest(TsdStartRequest):
    """Request for end-work — includes quantities + flags."""

    qty_complete: float = Field(default=0.0, ge=0)
    qty_scrapped: float = Field(default=0.0, ge=0)
    oper_complete: bool = False
    job_complete: bool = False


class TsdResponse(BaseModel):
    """Response from any TSD operation."""

    status: str  # "ok" or "error"
    message: str = ""
    actual_hours: Optional[float] = None
    stroj: Optional[str] = None
    wc: Optional[str] = None
